components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridPolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,422 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.entitlement.persistence;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.common.testng.WithRegistry;
+import org.wso2.carbon.identity.entitlement.dto.PolicyDTO;
+import org.wso2.carbon.identity.entitlement.dto.PolicyStoreDTO;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementConfigHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+
+import java.util.List;
+import java.util.Properties;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This class tests the behavior of the Hybrid Policy Persistence Manager class.
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridPolicyPersistenceManagerTest extends PolicyPersistenceManagerTest {
+
+    private JDBCPolicyPersistenceManager jdbcPolicyPersistenceManager;
+    private RegistryPolicyPersistenceManager registryPolicyPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        Properties storeProps = new Properties();
+        policyPersistenceManager = new HybridPolicyPersistenceManager();
+        policyPersistenceManager.init(storeProps);
+        jdbcPolicyPersistenceManager = new JDBCPolicyPersistenceManager();
+        registryPolicyPersistenceManager = new RegistryPolicyPersistenceManager();
+        registryPolicyPersistenceManager.init(storeProps);
+    }
+
+    @Test(priority = 13)
+    public void testAddPAPPolicyInDb() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+
+        PolicyDTO policyFromStorage = policyPersistenceManager.getPAPPolicy(samplePAPPolicy1.getPolicyId());
+        assertEquals(policyFromStorage.getPolicy(), samplePAPPolicy1.getPolicy());
+        assertEquals(policyFromStorage.getPolicyId(), samplePAPPolicy1.getPolicyId());
+        assertEquals(policyFromStorage.getVersion(), "1");
```

Comment:
Since these are already validated in the `testAddPAPPolicy()` better to validate these details against the `jdbcPolicyPersistenceManager` if we are to validate data here

Reply:
https://github.com/wso2/carbon-identity-framework/compare/2f503927491810d4d2cc8b7ea09b65f587525e2e..585e1901da74cee150023ab7584efca46ed666d1